### DISCUSSION TITLE
How to correctly integrate Google Sheet & FB Custom Audience

### DISCUSSION THREAD

**Discussion by PeterLangan** [merchantprotocol/theadslab#41] [User Guide Q&A]
Using the web hook we are generating hash data for the traffic to a clients website. We would like to retarget this traffic via a FB custom audience and have been attempting to create a ZAP to this effect. 

Zap permits us to create 1. New Custom Audience or 2. Add Email To Custom Audience. 

Neither of these permits us to utilize the hash data. How do I do this, unless I am missing something obvious? 

![image](https://user-images.githubusercontent.com/102678462/163358723-87978785-1b1b-4098-bfd8-8bcf7fc54ece.png)
![image (1)](https://user-images.githubusercontent.com/102678462/163358730-65dd98ae-892c-4686-a0a3-69cb66f312d3.png)
![image (2)](https://user-images.githubusercontent.com/102678462/163358733-d5adbc07-8238-4278-be4f-d03637a0d1ae.png)

**Comment by jonathonbyrdziak** [comment]
Here's what Facebook says about using hashed data.

[Customer File Custom Audiences](https://developers.facebook.com/docs/marketing-api/audiences/guides/custom-audiences)

> Hashing and Normalization for Multi-Key
> 
> You must hash your data as SHA256; we don't support other hashing mechanisms. This is required for all data except External Identifiers, App User IDs and Page Scoped User IDs. Before hashing, normalize your data.
> 
> Email addresses - Use key EMAIL. Trimming leading and trailing whitespace and convert all characters to lowercase.
> 
> Provide SHA256 values for normalized keys and HEX representations of this value, using lowercase for A through F. The hash function in PHP converts normalized email and phone number:
> 
> example : hash("sha256", "mary@example.com") f1904cf1a9d73a55fa5de0ac823c4403ded71afd4c3248d00bdcd0866552bb79

The hash that we send you via the pixel fire webhook event is MD5, which is not compatible with Facebook custom audiences. You'll need to use the SHA256 field found in the 'Consumer Email' data.

This value should be uploaded as an EMAIL into Facebook's custom audiences.

**Reply by PeterLangan** [reply]
Hi Jonathon. Thanks for this. I want to retarget a large portion of the traffic which visits the site, if I use the data from Consumer Email I am only retargeting the resolved identities.

The webhook cannot retrieve the SHA256. 

What can I do to get the SHA 256 of the non resolved identities? Or what identifier do I use for re targetting those people (non fully resolved) for which we don't have the SHA 256 email address. 

Can I retarget via webhook of device ID?
